src/index.js

```diff
@@ -1,6 +1,20 @@
 const Logger = require("bunyan");
 
-function findLogLevelNameForCategory(config, category, parentConfigMinLevel) {
+function categoryConfigHasMatchingConditional(categoryConfig, record) {
+  const { minLevel, ...configurationToMatch } = categoryConfig.conditional
+  let allPropertiesMatch = true;
+
+  for(const key of Object.keys(configurationToMatch)) {
+    if(!record || record[key] !== configurationToMatch[key]) {
+      allPropertiesMatch = false;
+      break;
+    }
+  }
```

Comment:
It might be slightly faster to iterate the entries so the value doesn't have to be searched, something like:

```
  for (const [key, value] of Object.entries(configurationToMatch)) {
    if(!record || record[key] !== value) {
      allPropertiesMatch = false;
      break;
    }
  }
```